### ISSUE TITLE
Flush will not finish and write the current block.

### ISSUE THREAD

**Issue by MichaelGehling** [MiloszKrajewski/K4os.Compression.LZ4#104]
**Description**
I am using LZ4Stream to encode data into a stream. I need to be able to force to finish a block and later keep on compressing the next block. Hence, I set ChainBlocks to false, and whenever I need to do a split, I call Flush() with the expectation that the compressor stream would finish the current block and write it to the target stream.
But instead, it does only flush the target stream and the unfinished block is not written.

**To reproduce**
```
var compressionStream = LZ4Stream.Encode(memoryStream, new LZ4EncoderSettings()
{
    ChainBlocks = false,
}, true);
// Write small amount of data
compressionStream .Flush()
//memoryStream.Position will still be 0

```

**Expected behavior**
I want that Flush actually writes data from the current block.

**Actual behavior**
Flush does not write any data, but just flushes the target stream (inner stream)

**Environment**
 - CPU: AMD 64 bit
 - OS: Windows
 - .NET: .Net 8
 - LZ4: 1.3.8

**Comment by MiloszKrajewski** [comment]
Yes. It is by design and you would need to do serious convincing that your use case is legitimate. 
Use `Close` instead to write all remaining data.

It is actually the same as `System.IO.Compression.DeflateStream` for roughly same reasons:

https://learn.microsoft.com/en-us/dotnet/api/system.io.compression.deflatestream.flush?view=net-8.0#system-io-compression-deflatestream-flush

There are reasons behind it, but it would require few paragraphs of text, so unless you insist I will skip explanation for now ;-)

**Comment by MiloszKrajewski** [comment]
As I said above, it is not a bug as this is by design, but think it can be added the same way "interactive" reading was added on read side. There is some value in sending data over the network to reader without closing stream.

I am accepting it as enhancement suggestion... not a bug ;-)
